src/providers/mdal/qgsmdalprovider.cpp

```diff
@@ -241,6 +241,22 @@ bool QgsMdalProvider::persistDatasetGroup( const QString &path,
   return false;
 }
 
+void QgsMdalProvider::reloadData()
+{
+  if ( mMeshH )
```

Comment:
I think you need to reset mNativeMesh, mTriangularMesh and mRendererCache

Reply:
mNativeMesh and mRendererCache is reset in the new reload() method.
mTriangularMesh will be updated when createRendrer() is called, no ?

Reply:
what is "new reload" method? I would just reset with NULL for all three